dataverse/orgsvc/C#-NETCore/BulkOperations/UpsertMultiple/Program.cs

```diff
@@ -0,0 +1,269 @@
+
+                        // TODO : Need to come up with a way to get all the IDs to delete
+                        string deleteJobStatus = Utility.BulkDeleteRecordsByIds(
+                            service: serviceClient,
+                            tableLogicalName: tableLogicalName,
+                            iDs: createdRecordIds,
+                            jobName: "Deleting records created by UpsertMultiple Sample.");
+
+                        Console.WriteLine($"\tBulk Delete status: {deleteJobStatus}");
+                    }
+
+
+                }
+
+                // Delete sample_example table
+                Utility.DeleteExampleTable(
+                    service: serviceClient,
+                    tableSchemaName: tableSchemaName);
+
+            }
+            catch (FaultException<OrganizationServiceFault> ex)
+            {
+                Console.WriteLine("The application terminated with an error.");
+                Console.WriteLine("Timestamp: {0}", ex.Detail.Timestamp);
+                Console.WriteLine("Code: {0}", ex.Detail.ErrorCode);
+                Console.WriteLine("Message: {0}", ex.Detail.Message);
+                Console.WriteLine("Inner Fault: {0}",
+                    null == ex.Detail.InnerFault ? "No Inner Fault" : "Has Inner Fault");
+            }
+            catch (TimeoutException ex)
+            {
+                Console.WriteLine("The application terminated with an error.");
+                Console.WriteLine("Message: {0}", ex.Message);
+                Console.WriteLine("Stack Trace: {0}", ex.StackTrace);
+                Console.WriteLine("Inner Fault: {0}",
+                ex.InnerException.Message ?? "No Inner Fault");
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine("The application terminated with an error.");
+                Console.WriteLine(ex.Message);
+
+                // Display the details of the inner exception.
+                if (ex.InnerException != null)
+                {
+                    Console.WriteLine(ex.InnerException.Message);
+
+                    if (ex.InnerException is FaultException<OrganizationServiceFault> fe)
+                    {
+                        Console.WriteLine("Timestamp: {0}", fe.Detail.Timestamp);
+                        Console.WriteLine("Code: {0}", fe.Detail.ErrorCode);
+                        Console.WriteLine("Message: {0}", fe.Detail.Message);
+                        Console.WriteLine("Trace: {0}", fe.Detail.TraceText);
+                        Console.WriteLine("Inner Fault: {0}",
+                            null == fe.Detail.InnerFault ? "No Inner Fault" : "Has Inner Fault");
+                    }
+                }
+            }
+        }
+
+        public static Guid[] CreateRecordsUtility(int numberOfRecords, string tableLogicalName, ServiceClient serviceClient, out EntityCollection entities)
```

Comment:
Rather than ServiceClient, use IOrganizationService.

ServiceClient implements IOrganizationService and we need to reinforce that everywhere.

We only use ServiceClient when using the [methods specific to ServiceClient](https://learn.microsoft.com/en-us/dotnet/api/microsoft.powerplatform.dataverse.client.serviceclient?view=dataverse-sdk-latest#methods) which are many but can't be used in plug-ins.

Reply:
Done